### PR TITLE
Archivemount restore working dir

### PR DESCRIPTION
**Pull request by clementFoyer**
This PR saves the current working directory and restores it after the topology has been loaded.

Solves #577 by ensuring the current directory does not change (at least not from a side effect of loading a topology).

### CODE REVIEW

File: utils/hwloc/misc.h [hunk 1]
```diff
@@ -325,6 +335,19 @@ hwloc_utils_enable_input_format(struct hwloc_topology *topology, unsigned long f
     err = hwloc_utils_enable_input_format(topology, flags, subdir, &sub_input_format, verbose, callname);
     if (!err)
       *input_format = sub_input_format;
+    else {
```
Comment (clementFoyer) [inline utils/hwloc/misc.h:338]: The 4 lines here depend on/are linked to PR #584

**Comment by clementFoyer** [comment]
The issue with the build is the same as [there](https://github.com/netblue30/firejail/issues/2696), i.e., CentOS 6 does not declare O_PATH which is required to use `fchdir`.

**Comment by bgoglin** [comment]
If you can make your code conditional with #ifdef O_PATH, I'd be fine with only having the issue on very old kernels. I am not even sure archivemount works on them anyway.

File: utils/hwloc/misc.h [hunk 2]
```diff
@@ -325,6 +335,19 @@ hwloc_utils_enable_input_format(struct hwloc_topology *topology, unsigned long f
...
+    }
+
+    err = fchdir(oldworkdir);
+    if (err) {
+      perror("Restoring current working directory");
+      close(oldworkdir);
+      return EXIT_FAILURE;
+    }
+
+    close(oldworkdir);
```
Comment (bgoglin) [inline utils/hwloc/misc.h:350]: Not sure that's enough. The goal is to keep a reference on the archivemount'ed path during topology_load(), which happens after this code. So you'd likely need to move fchdir+close after load(). That's a bit ugly but I guess we'd need to make oldworkdir a global variable initialized to -1, and that variable would need to be added to all users of "input_format" macros :/

**Review by bgoglin — COMMENTED** [review]
Can you actually merge the double single-commit message into a single single-commit message ? :)

**Comment by clementFoyer** [comment]
Sorry, I missed your last comment. I merged the two messages into a single one.

**Comment by bgoglin** [comment]
Thanks.

**Pull request merged**